sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+          rc = az_iot_hub_client_methods_response_get_publish_topic(
+              &client,
+              request->request_id,
+              status,
+              commands_response_topic,
+              sizeof(commands_response_topic),
+              NULL)))
+  {
+    printf("Unable to get twin document publish topic\n");
+    return rc;
+  }
+
+  printf("Status: %u\tPayload:", status);
+  char* payload_char = (char*)az_span_ptr(response);
+  if (payload_char != NULL)
+  {
+    for (int32_t i = 0; i < az_span_size(response); i++)
+    {
+      putchar(*(payload_char + i));
+    }
+  }
+  putchar('\n');
+
+  // Send the commands response
+  rc = mqtt_publish_message(commands_response_topic, response, 0);
+
+  printf("Sent response\n");
+
+  return rc;
+}
+
+// Build the JSON payload for the reported property
+static az_result build_confirmed_reported_property(
+    az_json_builder* json_builder,
+    az_span property_name,
+    double property_val,
+    int32_t ac,
+    int32_t av,
+    az_span ad)
+{
+  AZ_RETURN_IF_FAILED(az_json_builder_init(json_builder, AZ_SPAN_FROM_BUFFER(reported_property_payload), NULL));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_begin_object(json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, property_name));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_begin_object(json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, AZ_SPAN_FROM_STR("value")));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, (int32_t)property_val));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, AZ_SPAN_FROM_STR("ac")));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, ac));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, AZ_SPAN_FROM_STR("av")));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, av));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, AZ_SPAN_FROM_STR("ad")));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_string(json_builder, ad));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(json_builder));
+
+  return AZ_OK;
+}
+
+static az_result build_reported_property(
+    az_json_builder* json_builder,
+    az_span property_name,
+    double property_val)
+{
+  AZ_RETURN_IF_FAILED(az_json_builder_init(json_builder, AZ_SPAN_FROM_BUFFER(reported_property_payload), NULL));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_begin_object(json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(json_builder, property_name));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, (int32_t)property_val));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(json_builder));
+
+  return AZ_OK;
+}
+// Send the twin reported property to the service
+static int send_reported_temperature_property(double temp_value, int32_t version)
```

Comment:
>send_reported_temperature_property [](start = 11, length = 34)

(1) Are you sure you want to make temp_value a double here?  I get current limitations of JSON parser mean you can only get out a uint, but to my mind it's cleaner to have it go back as a uint too since otherwise I'm wondering why you did the type conversion.  Then when doubles are ready you update the parse&report.

(2) Do you document anyplace the caveat that we're going to parse out what could be a double as an int at the end of the day?  What would happen if customer sent a double to us anyway?

Reply:
I'd like to keep most things as double's so that when the update comes it's easier to sub in the new api.

Reply:
Ahhhh wait I think the change was put in in the last 12 hours:
https://github.com/Azure/azure-sdk-for-c/blob/a25ddc7e2ea821bf8b09c6ed2cfa59ce51e3cf81/sdk/inc/azure/core/az_json.h#L143

I'll update.

Reply:
Crap I have to update the API signatures for everything else too

Reply:
Sorry about that.
On the bright side, double support! :)